### PR TITLE
Feat: endpoint property support

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type** `Feature`

**Description** The PR adds endpoint property support ReactiveSearch components.

[📔  Notion](https://www.notion.so/appbase/endpoint-property-support-in-RS-acd0540c4a274b99b2b454540ec210ba)

https://www.loom.com/share/e09e4f68fd22442aad6c908306371e44
https://www.loom.com/share/259e327e014e475a98002e9383762238

https://github.com/appbaseio/reactivecore/pull/136
https://github.com/appbaseio/Docs/pull/349
https://github.com/appbaseio/appbase-js/pull/93

### CODE REVIEW

**Review by bietkul — CHANGES_REQUESTED** [review]
@mohdashraf010897 PR looks good 👍.
Can you also share the demos with SSR for Vue and React?

**Comment by mohdashraf010897** [comment]
> @mohdashraf010897 PR looks good 👍. Can you also share the demos with SSR for Vue and React?

https://www.loom.com/share/f6ec79f0fc1b438ca15b38f010b543e1

File: packages/vue/examples/with-ssr/nuxt.config.js [hunk 1]
```diff
@@ -30,14 +26,12 @@ export default {
...
-	modules: [
-	],
+	modules: [],
 
 	// Build Configuration: https://go.nuxtjs.dev/config-build
-	build: {
-	}
-}
+	build: {},
+	debug: true,
```
Comment (bietkul) [inline packages/vue/examples/with-ssr/nuxt.config.js]:
```suggestion
```

File: packages/vue/package.json [hunk 2]
```diff
@@ -37,7 +37,7 @@
 		"@appbaseio/reactivecore": "9.14.20",
 		"@appbaseio/vue-emotion": "0.4.4",
 		"@vue/babel-helper-vue-jsx-merge-props": "^1.0.0",
-		"appbase-js": "^5.0.0",
+		"appbase-js": "^5.1.0-alpha.3",
```
Comment (bietkul) [inline packages/vue/package.json]:
```suggestion
		"appbase-js": "^5.0.0",
```

File: packages/web/package.json [hunk 3]
```diff
@@ -40,7 +40,7 @@
 		"@appbaseio/reactivecore": "9.14.20",
 		"@emotion/core": "^10.0.28",
 		"@emotion/styled": "^10.0.27",
-		"appbase-js": "5.0.0",
+		"appbase-js": "5.1.0-alpha.3",
```
Comment (bietkul) [inline packages/web/package.json]:
```suggestion
		"appbase-js": "5.0.0",
```

File: packages/vue/src/server/index.js [hunk 4]
```diff
@@ -472,14 +497,17 @@ export default function initReactivesearch(componentCollection, searchState, set
 										};
 									}
 									timestamp[component] = res._timestamp;
+									const hitsObj = response.hits
```
Comment (bietkul) [inline packages/vue/src/server/index.js:508]: @mohdashraf010897 This change is not required, which seems specific to your endpoint example.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: packages/web/src/server/index.js [hunk 5]
```diff
@@ -94,20 +98,39 @@ export default function initReactivesearch(componentCollection, searchState, set
...
+				? settings.endpoint.headers
+				: {}),
 		};
+		let url
+			= settings.url && settings.url.trim() !== ''
+				? settings.url
+				: 'https://scalr.api.appbase.io';
+
+		let transformRequest = settings.transformRequest || null;
+		if (settings.enableAppbase && settings.endpoint instanceof Object) {
```
Comment (bietkul) [inline packages/web/src/server/index.js]:
```suggestion
		if (settings.enableAppbase && settings.endpoint && settings.endpoint instanceof Object) {
```

**Review by bietkul — CHANGES_REQUESTED** [review]

**Pull request merged**